File: server/app/schemas/wafer_map.py
from pydantic import BaseModel


class DieCell(BaseModel):
    x: int
    y: int
    bin: int


class WaferMapResponse(BaseModel):
    waferId: str
    dies: list[DieCell]
    maxX: int
    maxY: int
    minX: int
    minY: int


class BinCount(BaseModel):
    bin: int
    label: str
    count: int


class WaferStatsResponse(BaseModel):
    totalDies: int
    bin1Pass: int
    bin1Yield: float
    failCount: int


class BinDistributionResponse(BaseModel):
    bins: list[BinCount]

File: src/services/waferMap.ts
import { apiFetch } from './api'

export interface DieCell {
  x: number
  y: number
  bin: number
}

export interface WaferMapData {
  waferId: string
  dies: DieCell[]
  maxX: number
  maxY: number
  minX: number
  minY: number
}

export interface WaferStats {
  totalDies: number
  bin1Pass: number
  bin1Yield: number
  failCount: number
}

export interface BinCount {
  bin: number
  label: string
  count: number
}

export async function getWaferMap(waferDbId: number): Promise<WaferMapData> {
  return apiFetch(`/wafer-map/${waferDbId}`)
}

export async function getWaferStats(waferDbId: number): Promise<WaferStats> {
  return apiFetch(`/wafer-map/${waferDbId}/statistics`)
}

export async function getBinDistribution(waferDbId: number): Promise<{ bins: BinCount[] }> {
  return apiFetch(`/wafer-map/${waferDbId}/bin-distribution`)
}
